crates/fuel-core/src/database/state.rs

```diff
@@ -231,29 +232,26 @@ mod tests {
             }
 
             // then
-            let states_in_db: Vec<_> = database
+            let mut states_in_db: Vec<_> = database
                 .iter_all::<ContractsState>(None)
                 .collect::<Result<Vec<_>, _>>()
                 .unwrap()
                 .into_iter()
-                .map(|(key, value)| {
-                    let contract_id = *key.contract_id();
-                    let key = *key.state_key();
-
-                    ContractStateConfig {
-                        contract_id,
-                        key,
-                        value: value.0,
-                    }
-                })
                 .collect();
 
-            let original_state = state_groups
+            let mut original_state = state_groups
                 .into_iter()
                 .flatten()
-                .sorted()
+                .map(|v| (v.key, v.value))
                 .collect::<Vec<_>>();
 
+            states_in_db.sort_by(|(ak, av), (bk, bv)| {
+                (ak.as_ref(), av.as_ref()).cmp(&(bk.as_ref(), bv.as_ref()))
+            });
+            original_state.sort_by(|(ak, av), (bk, bv)| {
+                (ak.as_ref(), av.as_ref()).cmp(&(bk.as_ref(), bv.as_ref()))
+            });
```

Comment:
The previous code was much simpler, why we don't want to compare `ContractStateConfig`?

Reply:
The old version didn't compile, so it was quite easy for it to be simple. I've tried to improve this a bit in 1c6f607e79dd0c8.